### PR TITLE
Added a way to disable the maven cache

### PR DESCRIPTION
**Pull request by phillip-kruger**
main currently failing (even before this change) with:

`ERROR] Failures: 
[ERROR]   RepositoryMetadataResourceTest.should_support_repository_proxied_as_nexus_repository:40 1 expectation failed.
Response body doesn't match expectation.
Expected: is "/foo"
  Actual: ## repository-prefixes/2.0
/foo`

Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java [hunk 1]
```diff
@@ -37,9 +37,12 @@
     @ConfigProperty(name = "maven.cache.header.cache-control", defaultValue = "no-cache") // Check for etag everytime
     Instance<String> headerCacheControl;
 
+    @ConfigProperty(name = "maven.cache.enabled", defaultValue = "true")
+    Instance<Boolean> enabled;
+    
     @Override
     public void filter(ContainerRequestContext containerRequestContext) throws IOException {
-        if (isMavenGetRequest(containerRequestContext)) {
+        if (isMavenGetRequest(containerRequestContext) && enabled.get()) {
```
Comment (gastaldi) [inline src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java:45]: What about the filter method from `ContainerResponseFilter`? Should it also check for this flag as well?

Comment (phillip-kruger) [reply src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java:45]: Well, assuming we can enable/disable in runtime, we can still cache the values, but not read from it, then when we want we can enable and have a hot cache... but only of we can enable without restart.

**Comment by gastaldi** [comment]
Test fixed in https://github.com/quarkusio/registry.quarkus.io/commit/d13f35eab89c65f8dae0b2ea8a4467ad7e4b7ebd, sorry for the trouble

File: src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java (hunk 1 above)
Comment (gastaldi) [reply src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java:45]: I am not sure we can enable it without restarting. Last time I checked `ConfigSources` were immutable during runtime.

Comment (phillip-kruger) [reply src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java:45]: OK, this is a use case to have the option to not have that :) Let me also check there then . Hold on

**Comment by gastaldi** [comment]
I think it looks fine. Do you think we need to worry about the `@CacheResult` annotations (it caches the metadata descriptor)?

**Comment by phillip-kruger** [comment]
I do not think so because that will not be cached by CDN, right ?

**Comment by gastaldi** [comment]
I think it might fail if you're running more than 1 replica

**Comment by phillip-kruger** [comment]
OK true.... but that is a separate issue ?

**Comment by gastaldi** [comment]
Yes, let's merge this first. Thanks!

**Pull request merged**